### ISSUE TITLE
[Bug] Incompatibility with REI

### ISSUE THREAD

**Issue by Micalobia** [AppliedEnergistics/Applied-Energistics-2#5658]
### Describe the bug

Typing into the REI textbox while in an AE2 terminal has some weird issues

### How to reproduce the bug

- Start to type a word
- When `e` is pressed, it exits the inventory
- When `backspace` is pressed, it goes into the AE2 textbox instead of REI, without refocusing

**Environment**
 - Minecraft 1.17.1
 - Latest Fabric loader for 1.17.1
 
 Mirrored issue on AE2 github: (WIP)

### Expected behavior

Typing into the correct text box without interaction issues per which one is focused

### Additional details

Mirrored issue on REI: shedaniel/RoughlyEnoughItems#655

### Which minecraft version are you using?

1.17

### On which mod loaders does it happen?

Fabric

### Crash log

N/A

**Comment by praus1** [comment]
Been having this same issue. Didn’t realize the Backspace was moving the cursor to the AE2 search bar, thought it just stopped working. One workaround I’ve found for the backspacing is the Right Click Clear mod. It clears search boxes with a right click. Obviously doesn’t fix it but it helps until it’s fixed.

**Comment by shartte** [comment]
This has been fixed in 1.18 now due to changing how focus behavior works and will be further made obsolete by #5825, which will hide the AE2 search box if REI synchronized mode is enabled. But even if that mode is not enabled, the AE2 search box should no longer steal the focus in 1.18 at this point.

**Comment by red2blue** [comment]
It's still being funky in 1.18

Here is a bug report I received yesterday from a player of my modpack aqm2

```
https://paste.corebot.dev/edixufemen.md when using AE2 terminal (every terminal) multiple things broken, I cannot delete words on JEI, cannot move items from AE to Inventory with shift + click, typing "E" (in input box) causes modal close. (sorry for spamming, using template is quite harder than I think)
```

![image](https://user-images.githubusercontent.com/35115304/146833138-a17c9d3c-c209-4460-8c11-9d779c806e6e.png)
